### PR TITLE
[devscripts] Minor improvements that fix doc and support optionally applying patch

### PR DESCRIPTION
**Pull request by psathyan**
As part of this PR, there are two changes

- minor doc fixes
- support applying of patch optional.
- switching the default OpenShift version to `4.14.0`

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: ci_framework/roles/devscripts/tasks/sub_tasks/16_user.yml [hunk 1]
```diff
@@ -14,11 +14,9 @@
...
 # under the License.
 
-- name: Create the specified user.
-  become: true
-  ansible.builtin.user:
-    name: "{{ cifmw_devscripts_user }}"
-    state: present
+- name: Gather the user
+  ansible.builtin.set_fact:
+    cifmw_devscripts_user: "{{ ansible_user_id }}"
```
Comment (pablintino) [inline ci_framework/roles/devscripts/tasks/sub_tasks/16_user.yml:19]: Do we really need this fact? We can use ansible_user_id straightforward

Comment (cjeanner) [reply ci_framework/roles/devscripts/tasks/sub_tasks/16_user.yml:19]: there's also a non-null risk of ppl overriding this via a parameter (while using an "old" parameter file for instance). Might be better to drop it for good.

File: ci_framework/roles/devscripts/molecule/default/converge.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@
   hosts: all
 
   vars:
-    ansible_user_dir: "{{ lookup('env', 'HOME') }}"
+    ansible_user_dir: "{{ ansible_env.HOME }}"
```
Comment (dasm) [inline ci_framework/roles/devscripts/molecule/default/converge.yml:22]: Is it being moved from controller over to target?

[docs](https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-access-shell-environment-variables)

File: ci_framework/roles/devscripts/vars/main.yml [hunk 3]
```diff
@@ -47,7 +47,7 @@ cifmw_devscripts_config_defaults:
   master_disk: 100
   master_vcpu: 10
   num_workers: 0
-  num_extra_workers: 3
+  num_extra_workers: 0
```
Comment (dasm) [inline ci_framework/roles/devscripts/vars/main.yml:50]: Can you remind why this change?

Comment (psathyan) [reply ci_framework/roles/devscripts/vars/main.yml:50]: The idea is to use `libvirt_manager` for creating the OSP computes and use `devscripts` for deploying only the OCP control plane. 

Setting this value to `0` prevents `dev-scripts` from creating additional nodes that could be used for compute. Hence changing the default behavior.

File: ci_framework/roles/devscripts/molecule/default/converge.yml (hunk 2 above)
Comment (psathyan) [reply ci_framework/roles/devscripts/molecule/default/converge.yml:22]: Initial design was to run this on the controller however it has changed for `reproducer` hence replacing all `lookup` to enable supporting of remote execution also.

**Comment by cjeanner** [comment]
/approve
allowing the execution against a remote node has been in the plans for some time already - so yep, replacing lookup() is a nice move, thank you!
Improving doc is also good :).

**Review by raukadah — APPROVED** [review]

File: ci_framework/roles/devscripts/molecule/default/converge.yml (hunk 2 above)
Comment (dasm) [reply ci_framework/roles/devscripts/molecule/default/converge.yml:22]: ack, thx

File: ci_framework/roles/devscripts/vars/main.yml (hunk 3 above)
Comment (dasm) [reply ci_framework/roles/devscripts/vars/main.yml:50]: thanks for the explanation
